pkg/restclient/common.go

```diff
@@ -163,3 +174,34 @@ func (c *Client) do(method string, url string, data interface{}, expectStatusCod
 func (c *Client) Ping() error {
 	return c.get("/ping", nil, http.StatusOK, nil)
 }
+
+func (c *Client) setAgent(req *http.Request) {
+	req.Header.Set("User-Agent",
+		fmt.Sprintf("%v (Go-http-client/1.1)", c.agent))
+}
+
+// ExtendAgent adds the given string to the client's user agent
+// by prefixing it to the existing agent information.
+// This allows client programs to identify themselves more than
+// just something using the rest client api.
+func (c *Client) ExtendAgent(a string) {
+	// new additions to the agent identifier go to the
+	// beginning of the string. It is meant to read like:
+	// foo (based on) bar (based on) baz, etc...
+	c.agent = fmt.Sprintf("%v %v", a, c.agent)
+}
+
+func (c *Client) sendOriginArgs(req *http.Request) {
+	if len(c.originArgs) != 0 {
+		req.Header.Set("X-Gluster-Origin-Args",
```

Comment:
This is nice. is this logged? I don't see this detail in output @prashanthpai posted https://github.com/gluster/glusterd2/pull/978#issuecomment-404738074

Reply:
It should be, but only if the client sends it. I'm not exactly sure exactly how @prashanthpai got his examples, but when I incorporate the other feedback for these changes I'll include examples of the logging as well.

Reply:
My bad. I think I did `make glusterd2` as I usually do. I can see it after I recompiled the CLI.

```log
INFO[2018-07-17 10:03:37.209081] HTTP Request                                  client="glustercli/v4.1.0-97.git987fac1 GlusterD2-rest-client/v4.1.0-97.git987fac1 (Go-http-client/1.1)" method=GET origin="1:\"glustercli peer list\"" reqid=817c7683-37dd-4b04-a589-16136deaf766 source="[request_logging.go:29:middleware.LogRequest.func1]" subsys=rest url=/v1/peers
INFO[2018-07-17 10:03:37.209832] 127.0.0.1 - - [17/Jul/2018:10:03:37 +0530] "GET /v1/peers HTTP/1.1" 401 71 "" "glustercli/v4.1.0-97.git987fac1 GlusterD2-rest-client/v4.1.0-97.git987fac1 (Go-http-client/1.1)"  origin="1:\"glustercli peer list\"" reqid=817c7683-37dd-4b04-a589-16136deaf766 subsys=rest
```

Reply:
nice, thanks for the update